controllers/territory_controller/territory_controller.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
Is this import needed? Feels a bit suspect

Reply:
Mypy won't let you have a classmethod which returns its own class in the type signature without it. See line 36

Reply:
The more usual (though perhaps now "legacy") way to solve that would be to quote the annotation.
I'd not realised that this future import was available to us though :)

Reply:
The feature is added in 3.10

Reply:
Ah yeah I've seen it as quoted, and I knew newer versions supported just specifying it. Yeah let's keep the future import then :+1: